python/python/lancedb/_lancedb.pyi

```diff
@@ -0,0 +1,12 @@
+from typing import Optional
+
+class Connection(object):
+    pass
+
+def connect(
+    uri: str,
+    api_key: Optional[str],
+    region: Optional[str],
+    host_override: Optional[str],
+    read_consistency_interval: Optional[float],
+) -> Connection: ...
```

Comment:
Doesn't this return a Future?

Reply:
I changed it to `async def` which should imply the future.  I'm on the fence about these having these pyi files at all and curious for your opinion.  These methods aren't exposed to the user so this is only for our own internal understanding.

Reply:
For internal only things, I don't think we have to write them. Sometimes we might publicly expose them though, if the Python wrapper is redundant.

If I add actual mypy checks to the repo, my opinion on this might change.